server/rtcd.go

```diff
@@ -30,6 +30,85 @@ func (p *Plugin) getStoredRTCDConfig() (rtcd.ClientConfig, error) {
 	return cfg, nil
 }
 
+func (p *Plugin) registerRTCDClient(cfg rtcd.ClientConfig) (rtcd.ClientConfig, error) {
+	client, err := rtcd.NewClient(cfg)
+	if err != nil {
+		return cfg, fmt.Errorf("failed to create rtcd client: %w", err)
+	}
+	defer client.Close()
+
+	cfg.AuthKey, err = client.Register(cfg.ClientID)
+	if err != nil {
+		return cfg, fmt.Errorf("failed to register rtcd client: %w", err)
+	}
+
+	cfgData, err := json.Marshal(&cfg)
+	if err != nil {
+		return cfg, fmt.Errorf("failed to marshal rtcd client config: %w", err)
+	}
+
+	if err := p.API.KVSet(rtcdConfigKey, cfgData); err != nil {
+		return cfg, fmt.Errorf("failed to store rtcd client config: %w", err)
+	}
+
+	p.LogDebug("rtcd client registered successfully", "clientID", cfg.ClientID)
+
+	return cfg, nil
+}
+
+func (p *Plugin) newRTCDClient(rtcdURL string) (*rtcd.Client, error) {
+	clientCfg, err := p.getRTCDClientConfig(rtcdURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get rtcd client config: %w", err)
+	}
+
+	client, err := rtcd.NewClient(clientCfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create rtcd client: %w", err)
+	}
+
+	err = client.Connect()
+	if err == nil {
+		return client, nil
+	}
+	defer client.Close()
+
+	// If connecting fails we attempt to re-register once as the rtcd instance may
+	// have restarted, potentially losing stored credentials.
```

Comment:
Should we make a longer pause then, maybe wait long enough for an rtcd instance to restart? (Is that even a worry?)

Reply:
The assumption here is that `rtcd` restarted before plugin activation and we'd be failing to authenticate because the client would be effectively unregistered again at that point. The case of `rtcd` restarting while the plugin is connected will be handled in https://mattermost.atlassian.net/browse/MM-43508.